lib/sass/script/functions.rb

```diff
@@ -2393,13 +2456,17 @@ def global_variable_exists(name)
     #
     # @overload function_exists($name)
     #   @param name [Sass::Script::Value::String] The name of the function to
-    #     check.
+    #     check or a callable.
```

Comment:
I'm not a big fan of allowing this to take a callable. I get that it makes more existing code work as-is, but once the transition is done it's a weird behavioral corner case that exists forevermore. Making sure every corner of the APIs are compatible across functions and references is something I don't want to have to worry about; I think we should just have people do `type-of()` checks manually if they want to check "this function name exists or this is a callable".

Honestly it might also be simpler to just make `function-reference()` take only strings as well.